src/parser/switch.js

```diff
@@ -34,17 +34,17 @@ module.exports = {
     // DETECT SWITCH MODE
     let expect = null;
     const result = this.node("block");
+    this.expect(["{", ":"]);
```

Comment:
expect was running before only if token != '{' && != ':' now it runs every time, even on right tokens, no need to call this extra check, make it slower, please can you put as before